api/src/main/java/edu/cornell/mannlib/vitro/webapp/auth/permissions/EntityPermission.java

```diff
@@ -0,0 +1,274 @@
+/* $This file is distributed under the terms of the license in LICENSE$ */
+
+package edu.cornell.mannlib.vitro.webapp.auth.permissions;
+
+import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.RequestedAction;
+import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.propstmt.AbstractDataPropertyStatementAction;
+import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.propstmt.AbstractObjectPropertyStatementAction;
+import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.propstmt.AbstractPropertyStatementAction;
+import edu.cornell.mannlib.vitro.webapp.beans.DataProperty;
+import edu.cornell.mannlib.vitro.webapp.beans.FauxProperty;
+import edu.cornell.mannlib.vitro.webapp.beans.ObjectProperty;
+import edu.cornell.mannlib.vitro.webapp.beans.Property;
+import edu.cornell.mannlib.vitro.webapp.dao.PropertyDao;
+import edu.cornell.mannlib.vitro.webapp.dao.VitroVocabulary;
+import edu.cornell.mannlib.vitro.webapp.dao.WebappDaoFactory;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ContextModelAccess;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ModelNames;
+import edu.cornell.mannlib.vitro.webapp.utils.RelationshipCheckerRegistry;
+import org.apache.jena.ontology.OntModel;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.shared.Lock;
+import org.apache.jena.vocabulary.RDF;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+
+public abstract class EntityPermission extends Permission {
+    protected EntityPermission(String uri) {
+        super(uri);
+    }
+
+    /**
+     * Static fields for all EntityPermissions
+     */
+    private static final Map<String, EntityPermission> allInstances = new HashMap<String, EntityPermission>();
+    private static ContextModelAccess ctxModels = null;
+
+    /**
+     * Instance fields for each EntityPermission
+     */
+    private final Set<PropertyDao.FullPropertyKey> authorizedKeys = new HashSet<>();
+    private final Set<String> authorizedResources = new HashSet<>();
```

Comment:
This is because authorisation checks on resources in the existing code always results in authorisation being granted, and so this branch reflects that behaviour.

It's an open question as to whether we should fully implement resource checks - I can see cases where it would be required, but it may also cause a significant issue for upgrading existing sites.

Reply:
This isn't a reason to keep a private final variable whose value is never accessed. If we need it in the future, we'll add it.